Mecha resprite and Vim rework

kipdotnet opened the pull request:
<!-- Impstation note: there's no need to read all the official contributing guidelines,
but please DON'T combine upstream changes with your own changes.
Make separate pull requests for separate changes. -->
<!-- Guidelines: https://docs.spacestation14.io/en/getting-started/pr-guideline -->
mom said its my turn to resprite things
resprites the Ripley and Vim, and allows goblins to pilot them. please dont be evil about this so I dont have to nerf it
**Changelog**
<!-- Impstation note: we have our own AUTOMATIC changelog now, so please DO use this section! -->
<!-- Add a Changelog entry to make players aware of new features or changes that could affect gameplay.
Make sure to read the guidelines and take this Changelog template out of the comment block in order for it to show up.
Changelog must have a :cl: symbol, so the bot recognizes the changes and adds them to the game's changelog. -->
:cl:
- tweak: The Vim and Ripley now have new sprites.
- tweak: Goblins can now pilot a Vim.
- tweak: the Ripley APLU now has a slightly more decent melee attack.

kipdotnet commented:
Thats fair. Should I drop the resprite as well, then?

mutanteggs commented:
> Thats fair. Should I drop the resprite as well, then?

unless the module is like, a huge part of the resprite, no way, don't waste that work you did. the headlights and the resprite are cool!

hivehum commented:
dropping the module slot is fine but i think the resprite and allowing vims to be piloted by goblins and apids is fine personally. i dont think this is a power creep situation (what is it... power creeping...? what items are becoming less useful by reworking vims...?)

hivehum commented:
oh actually you only tagged goblins as vim pilots and not apids. did you mean to tag both? i know that was in the previous pr

kipdotnet commented:
Theyre actually already pilotable by apids since they imhereted that when I based their original version off of punpun. Punpun can also pilot a vim apparently. And yeah I should make note of the ripley damage change I forgot to put it in the pr notes

iaada commented:
As a rule of thumb if you recognize people can be evil with something, you probably shouldn't give it to them. Mods have enough to moderate, and something as difficult to notice as a Vim carrying 8 crates is going to get missed.

I think there ought to be a whitelist for modules equippable by vim, and specific vim modules. A hydraulic clamp with only one slot, the H.O.N.K. horn but without all the dangerous sounds. Small things that thematically fit the size and cost of a vim.

kipdotnet commented:
Ok I cut the ripley melee down to more reasonable numbers, removed the vims module slot and fixed the licensing. this should be good to look over again if anyone wants to yell at me more

Darkmajia reviewed (approved):
if we have a ripley melee meta i'm coming for you

kipdotnet commented:
If people use the ripley more its going to be because the sprite is so niceys. I hope. Its like a 10% buff or something
